### DISCUSSION TITLE
undefined method `disable!'

### DISCUSSION THREAD

**Discussion by moffat** [Homebrew/discussions#653] [Casks]
Hi

I've been maintaining a tap for the 'cdo' formula, a scientific package that was available in homebrew-science. Happily, it is now back in 'core', so there's no need for my tap anymore. 

I can of course untap my own tap/reinstall cdo and be done with it, but I'm not sure what happens with the users that are already tapped and using my version of 'cdo'. Reading the docs, it seems like the way to do this was to add something like:

`disable! date: "2020-01-20", because: "it is now in homebrew core. Please reinstall it as follows:\nbrew untap moffat/sciencebits\nbrew install cdo\n"`

to my formula, which I did here:

<https://github.com/moffat/homebrew-sciencebits/blob/master/Formula/cdo.rb>

when I try to upgrade my system (I'm trying to see what a user tapped to my repository would see) I get:

```
Error: Failed to import: /usr/local/Homebrew/Library/Taps/moffat/homebrew-sciencebits/Formula/cdo.rb
cdo: undefined method `disable!' for Formulary::FormulaNamespace1bcaa3bd4c9988650288929731827977:Module
```

so obviously I'm doing something wrong. Any ideas on how to tell any users tapped to my repository "stop using this and go to core". Delete the formula? something else?

Thanks.

**Comment by Rylan12** [comment]
The `disable!` method needs to be inside the formula class. See [`balance.rb`](https://github.com/Homebrew/homebrew-core/blob/862c623903330fd39465b0b5a5bd0e01d7691055/Formula/balance.rb#L19) for an example.

However, the better way to handle things like this is to use the `tap_migrations.json` file. See [this link](https://docs.brew.sh/Migrating-A-Formula-To-A-Tap) for more detailed instructions.

**Reply by moffat** [reply]
Thanks @Rylan12, rookie mistake. Fixed now.
